app/javascript/schools/courses/components/curriculum_editor/CurriculumEditor__TargetGroupEditor.re

```diff
@@ -130,6 +136,68 @@ let booleanButtonClasses = selected => {
 let formClasses = value =>
   value ? "drawer-right-form w-full opacity-50" : "drawer-right-form w-full";
 
+let handleErrorCB = (send, _) => send(UpdateSaving);
+
+let handleResponseCB =
+    (state, levelId, targetGroup, updateTargetGroupsCB, json) => {
+  let id = json |> Json.Decode.(field("id", string));
+  let sortIndex = json |> Json.Decode.(field("sortIndex", int));
+  let newTargetGroup =
+    TargetGroup.create(
+      id,
+      state.name,
+      Some(state.description),
+      state.milestone,
+      levelId,
+      sortIndex,
+      state.isArchived,
+    );
+  switch (targetGroup) {
+  | Some(_) =>
+    Notification.success("Success", "Target Group updated successfully")
+  | None =>
+    Notification.success("Success", "Target Group created successfully")
+  };
+  updateTargetGroupsCB(newTargetGroup);
+};
+
+let createTargetGroup =
+    (state, send, targetGroup, updateTargetGroupsCB, currentLevelId) => {
+  send(UpdateSaving);
+  let levelId =
+    state.levelId->Belt.Option.mapWithDefault(currentLevelId, l => l);
+  let payload = setPayload(state, levelId);
+  let url = "/school/levels/" ++ levelId ++ "/target_groups";
+  Api.create(
+    url,
+    payload,
+    handleResponseCB(state, levelId, targetGroup, updateTargetGroupsCB),
+    handleErrorCB(send),
+  );
+};
+
+let updateTargetGroup =
+    (
+      state,
+      send,
+      targetGroup,
+      updateTargetGroupsCB,
+      currentLevelId,
+      targetGroupId,
+    ) => {
+  send(UpdateSaving);
+  let levelId =
+    state.levelId->Belt.Option.mapWithDefault(currentLevelId, l => l);
```

Comment:
@harigopal At the time of creation, the level will be auto selected. 
When trying to move from one level to another if they clear the level their is no point in saving unless they have changed the title or description. its a bit complicated, Will have to dynamically compute dirty in-order to avoid unnecessary network request. I think its best to skip, I don't see a pain point here. :smile: